File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let l = prices.length;
    let curr = 0;
    let max = 0;
    for(let i = 0;i<l-1;i++){
        let profit = prices[i+1] - prices[i];
        curr += profit;
        if(curr < 0 ){
            curr = 0;
        }else{
            if(curr>max){
                max = curr;
            }
        }
    }
    return max;
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize){
    int max = 0;
    int curr = 0;
    for(int i = 0;i<pricesSize-1;i++){
        int profit = prices[i+1]-prices[i];
        curr += profit;
        if(curr<0){
            curr = 0;
        }else{
            if(curr>max){
                max = curr;
            }
        }
    }
    return max;
}